src/races.md

```diff
@@ -1,60 +1,103 @@
+<!--
 # Data Races and Race Conditions
+-->
 
+# データ競合と競合状態
+
+<!--
 Safe Rust guarantees an absence of data races, which are defined as:
+-->
+
+安全な Rust では、データ競合が存在しないことが保証されています。
+データ競合は、以下のように定義されています。
 
+<!--
 * two or more threads concurrently accessing a location of memory
 * one of them is a write
 * one of them is unsynchronized
+-->
 
+* 2 以上のスレッドが並行にメモリ上の場所にアクセスしている
+* この内 1 つは書き込み
+* この内 1 つは非同期
+
+<!--
 A data race has Undefined Behavior, and is therefore impossible to perform
 in Safe Rust. Data races are *mostly* prevented through rust's ownership system:
 it's impossible to alias a mutable reference, so it's impossible to perform a
 data race. Interior mutability makes this more complicated, which is largely why
 we have the Send and Sync traits (see below).
+-->
+
+データ競合は未定義動作を含み、そしてそれ故に安全な Rust で発生させることは不可能です。
+データ競合は Rust の所有権システムによって*ほとんど*防がれています。可変参照の
+エイリアスを生成することは不可能ですから、データ競合を起こすことは不可能です。
+内部可変性はこれをもっと複雑にします。これが、 Send トレイトと Sync トレイトが
+何故存在するかということの主な理由です (以下を見てください) 。
 
+<!--
 **However Rust does not prevent general race conditions.**
+-->
 
+**しかしながら Rust は、一般的な競合状態を防ぎません。**
+
+<!--
 This is pretty fundamentally impossible, and probably honestly undesirable. Your
 hardware is racy, your OS is racy, the other programs on your computer are racy,
 and the world this all runs in is racy. Any system that could genuinely claim to
 prevent *all* race conditions would be pretty awful to use, if not just
 incorrect.
+-->
+
+これは根本的に不可能で、そして多分本当に望まれていないものです。ハードウェアは
+競合状態を起こし、 OS も競合状態を起こし、コンピュータの他のプログラムも競合状態を起こし、
+そして世界中にある全てのプログラムも競合状態を起こします。どんなシステムでも、
+*全ての*競合状態を防げると喧伝しているようなものは、本当に使いづらいものとなるでしょう。
+たとえ正しいものだとしても。
```

Comment:
・・・ようなものは、単に間違っているだけでなく、本当に使いづらいものとなるでしょう。

Since it's fundamentally impossible, if a system claims it could prevent races at all, such a system would be incorrect *and* pretty hard to use.

Reply:
:+1: